WooCommerce/Classes/ViewRelated/Orders/Order Details/Receipts/ReceiptViewController.swift

```diff
@@ -51,15 +51,33 @@ final class ReceiptViewController: UIViewController {
     }
 
     @objc private func shareReceipt() {
+        ServiceLocator.analytics.track(event: .InPersonPayments.receiptEmailTapped(countryCode: nil,
+                                                                                   cardReaderModel: nil,
+                                                                                   source: .backend))
         guard let url = URL(string: viewModel.receiptURLString) else {
             return
         }
         let activityViewController = UIActivityViewController(activityItems: [url],
                                                 applicationActivities: nil)
+        activityViewController.completionWithItemsHandler = { [weak self] _, success, _, error in
+            if let error = error {
+                ServiceLocator.analytics.track(event: .InPersonPayments.receiptEmailFailed(error: error, source: .backend))
+                DDLogError("Failed to share receipt for orderID \(String(describing: self?.viewModel.orderID)). Error: \(error)")
+            }
+            switch success {
+            case true:
+                ServiceLocator.analytics.track(event: .InPersonPayments.receiptEmailSuccess(countryCode: nil, cardReaderModel: nil, source: .backend))
+            case false:
+                ServiceLocator.analytics.track(event: .InPersonPayments.receiptEmailCanceled(countryCode: nil, cardReaderModel: nil, source: .backend))
+            }
+        }
```

Comment:
When we track errors both in share and print actions, we'll also log the `success` or `completion` event. This is done on purpose to understand "recoverability": If an event failed, but then we tracked success on completion, the merchant could recover. If we track fail + !success then they couldn't.